### ISSUE TITLE
Add `plugin-list --remote`

### ISSUE THREAD

**Issue by danhper** [asdf-vm/asdf#211]
@doughsay
> Also, I like your ideas for the asdf install and asdf plugin-list --remote commands; I think we should finish this PR as is, but I would be happy to try implementing those ideas next.

I think this will be really helpful. A few points we might think about:

* Maybe `asdf plugin-list-all` would be better, as we currently have `asdf list-all` command. I personally prefer the flag approach, but I think we should try to stay consistent. We are not at 1.0 yet, so we can still choose to switch to flags later on.
* Having a huge list of plugins can be hard to parse, maybe we could add some information to each plugin, in order to have something easier to read
  * Add a `display-name` to each plugin. I think proper casing and spaces help visual grepping a lot  
  * Add a `category` to each plugin e.g. `language`, `package-manager`, `database`, `other`, and allow something like `asdf plugin-list-all --category=language` which would only show language
  * Add a `description` to each plugin, so that people not familiar with the language/database know more or less what it is about

Of course, adding all of this could take a while, and is not really fun, so we can probably add as we go. 

@asdf-vm/maintainers @doughsay What do you think?

**Comment by Stratus3D** [comment]
I think we should stick with flags as you suggested. 

I don't know that we have to necessarily add support for category and description metadata for plugins. I know what each of my plugins is used for since I installed them myself for specific reasons. Now with the plugin repository someone might not know all the details of the plugin they just installed since it's much easier. They still need to know the plugin they just installed isn't maintained by @asdf-vm/maintainers (aside from the 4 original plugins) and that they should consult the creator of that plugin when dealing with issues. We've had a decent amount of issues opened in the asdf project that actually should have been opened in one of the plugin repos, so maybe if we add metadata we should have a project_url field that's required in all plugins. They we could show it in certain scenarios (like listing the plugins, showing help for a plugin, etc...). This would hopefully reduce the of plugin issues that mistakenly opened in the asdf project.

What do you guys think? Does this make sense?

**Comment by danhper** [comment]
I will close this issue, as it has been inactive for a long time now.
If there is still interest, please feel free to open a new issue.
Thank you!
